frontend/packages/topology/src/utils/application-utils.ts

```diff
@@ -215,11 +225,17 @@ export const cleanUpWorkload = async (
     case StatefulSetModel.kind:
     case JobModel.kind:
     case CronJobModel.kind:
-      deleteRequest(modelFor(resource.kind), resource);
+      deleteRequest(resourceModel, resource);
+      break;
+    case EventingBrokerModel.kind:
+    case EventingChannelModel.kind:
+    case EventingIMCModel.kind:
+    case EventingKafkaChannelModel.kind:
```

Comment:
Eventing channels are dynamically registered and there could be more number of channels introduced in knative eventing, so instead of hard coding it I suggest use `useChannelModels` hook to get the installed channels models Kind and we need to include it here.

Eg:  The above code dint work, when I installed Nats Channel (ref: https://github.com/knative-sandbox/eventing-natss/tree/release-1.1/config)

![image](https://user-images.githubusercontent.com/9964343/148901895-a7417cce-419e-4270-b55d-8b8471d10b81.png)